docs/secure_coding_and_deployment_hardening/ssl.md

```diff
@@ -94,41 +93,96 @@ However, please note that the `ssl_crl_cache` module does not actually cache the
 
 ## Selecting protocol versions and ciphers
 
-In both clients and servers it is recommended to review the enabled TLS protocol versions and cipher suites, disabling weaker values that are not strictly required for interoperability. Additionally, in servers the cipher suites should be listed in order of preference (typically stronger ciphers should be listed first, and weaker ciphers included for compatibility reasons should be last), and the `honor_cipher_order` option should be set to true.
+Recent versions of Erlang/OTP disable most weak, legacy SSL/TLS protocol versions and cipher suites. For instance, Erlang/OTP 24 receives an 'A' score on the [Qualys SSL Labs 'SSL Server Test'](https://www.ssllabs.com/ssltest/), without any further tuning.
+
+Further hardening of the TLS parameters to comply with the Mozilla '[Server Side TLS](https://wiki.mozilla.org/Security/Server_Side_TLS)' "Intermediate compatibility" recommendations can be achieved as described below. These recommendations were written for servers, but the same settings may be used for client-side hardening, depending on the configuration of the TLS server(s) the client is expected to connect to.
 
 ```erlang
 %% Erlang
-Versions = ['tlsv1.2'],
-
-%% Start with TLS 1.2 defaults
-CipherSuites0 = ssl:cipher_suites(default, 'tlsv1.2'),
-CipherSuites = ssl:filter_cipher_suites(CipherSuites0, [
-    %% Only ECDHE key exchange, for forward secrecy
-    {key_exchange, fun
-        (ecdhe_ecdsa) -> true;
-        (ecdhe_rsa) -> true;
-        (_) -> false
-    end},
-    %% Exclude SHA1
-    {mac, fun
-        (sha) -> false;
-        (_) -> true
-    end}]).
+
+PreferredCiphers = [
+  %% Cipher suites (TLS 1.3): TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
+  #{cipher => aes_128_gcm, key_exchange => any, mac => aead, prf => sha256},
+  #{cipher => aes_256_gcm, key_exchange => any, mac => aead, prf => sha384},
+  #{cipher => chacha20_poly1305, key_exchange => any, mac => aead, prf => sha256},
```

Comment:
I believe this requires reasonably modern OTP version (24?).

Should it also be mentioned in the guide? I suspect OTP 22 simply won't accept the format.